rfcs/000-consolidate-amundsen-microservices.md

```diff
@@ -0,0 +1,154 @@
+- Feature Name: consolidate_amundsen_microservices
+- Start Date: 2021-03-23
+- RFC PR: [amundsen-io/rfcs#30](https://github.com/amundsen-io/rfcs/pull/30)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# Consolidate Amundsen Microservices
+
+## Summary
+
+The idea is to (eventually) deprecate the split of `amundsenmetadatalibrary`, `amundsenfrontendlibrary`, `amundsensearchlibrary` and `amundsencommon` repositories, and merge them all into one mono repository. The split will instead be based on the backend and frontend.
+
+Following will be the repositories (and packages) of Amundsen after the change.
+- amundsen (frontend, metadata, search, and common repositories)
+- amundsendatabuilder (Same as today's databuilder repository)
+
+There will not be a separate `amundsenmetadatalibrary`, `amundsenfrontendlibrary`, `amundsensearchlibrary` or `amundsencommon` packages.
+
+A few reasons why I am keeping the `amundsendatabuilder` out of the monorepo:
+- databuilder is a completely separate layer not directly related to the Amundsen project. If I am using Apache Atlas or GCP or any other such proxies, I'd not want to have databuilder code in my repo. 
+- databuilder is the only piece which will have a lot of integrations, so keeping it separate would reduce the noise from the actual amundesn repo/package. This will enable us to move really fast when introducing new integrations in databuilder.
+
+**No data will be lost from the database or search engine, as we will not be changing any models or data structure.**
+
+## Motivation
+
+At the moment, the Amundsen project is divided into multiple very thin backends or proxies, which are backed by other projects/databases/search services like Neo4j, Atlas, Neptune, etc. 
+
+Following are the Python/Flask based proxies powers the Amundsen project:
+- amundsenfrontendlibrary
+- amundsenmetadatalibrary
+- amundsensearchlibrary
+- amundsencommon 
+
+This architecture creates the following issues and is making it hard for companies to adopt Amundsen, and to customize it accordingly. 
+I've also added a note under each item explaining how this change will solve this issue.  
+
+#### 1: Dependency Management
+Python packages are pretty much the same for each proxy, which makes it hard to sync across all the above repositories. As a result most of the time, each proxy is running its own version of the dependency. 
+
+A few examples:
+
+| Package             | amundsenmetadata   | amundsenfrontend  |
+| ------------------  |:------------------:| -----------------:|
+| amundsen-common     | >=0.8.1            | ==0.6.0           |
+| flake8              | ==3.5.0            | ==3.8.4           |
+| flake8-tidy-imports | ==1.1.0            | ==4.2.1           |
+
+and many more... 
+
+*This change will put all the packages in just one place, making it easier to manage dependencies and no multiple variants of the same package.*
```

Comment:
Got confused by this, you mean we keep the separate packages or not? You meant `"merge" all the packages in just one place`?